keps/sig-apps/3017-pod-healthy-policy-for-pdb/README.md

```diff
@@ -0,0 +1,429 @@
+
+### Changes to the eviction API
+
+The eviction API will be updated to use `podHealthyPolicy` to determine whether
+a pod which is Running but not Ready can be evicted regardless of the value of
+`disruptionsAllowed`. This will only be a behavioral change when users have specified
+a `podHealthyPolicy`, and will not require the actual API to change.
+
+### Test Plan
+
+- Unit and integration tests covering:
+  - The current behavior stays unchanged when the policy is not specified.
+  - Correct behavior for both policies in both the disruption controller and
+    the eviction API.
+  - Feature gate disablement.
+- Verify passing existing E2E and conformance tests for PDBs and Eviction.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature gate disabled by default.
+- Unit and integration tests passing.
+
+#### Beta
+
+- Feature gate enabled by default.
+- Existing E2E and conformance tests passing.
+
+#### GA
+
+- Every bug report is fixed.
+- The disruption controller and the eviction API ignores the feature gate.
+
+#### Deprecation
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+No changes required for existing cluster to use the enhancement.
+
+### Version Skew Strategy
+
+This feature doesn't depend on the version for nodes.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: PodDisruptionBudgetPodHealthyPolicy
+    - Components depending on the feature gate:
+      - kube-controller-manager
+      - kube-apiserver
+- [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+###### Does enabling the feature change any default behavior?
+
+No, the behavior is only changed when users specify the `podHealthyPolicy` in
+the PodDisruptionBudget spec.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, in that case the disruption controller and the eviction API will just
+use the default legacy behavior.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The disruption controller and the eviction API will again start using
+the `PodHealthyPolicy` if provided on a PDB.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, there will be tests at unit and integration level.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+It does not change the default behavior. Users will have to specify a policy
+on the PDB for behavior to be affected.
+
+###### What specific metrics should inform a rollback?
+
+N/A
```

Comment:
I can easily imagine a buggy implementatio and e.g. disruption-controller crashing if the field is set or sth.

Reply:
Yeah, added more details here.